code/game/objects/items/devices/radio/radio_jammer.dm

```diff
@@ -0,0 +1,192 @@
+var/global/list/obj/item/device/radio_jammer/radio_jammer_list = list()
+
+#define JAMMING_SILENCE_SEVERITY 100
+
+/obj/item/device/radio_jammer
+	name = "radio jammer"
+	desc = "A device used to jam radio communications. Requires a power cell to function."
+	icon_state = "radio_jammer0"
+	flags = FPRINT
+	slot_flags = SLOT_BELT
+	w_class = W_CLASS_SMALL
+	item_state = "electronic"
+	starting_materials = list(MAT_IRON = 500, MAT_GLASS = 100)
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_PLASTIC
+	origin_tech = Tc_MAGNETS + "=3;" + Tc_ENGINEERING + "=4;" + Tc_MATERIALS + "=4;" + Tc_PROGRAMMING + "=3;" + Tc_SYNDICATE + "=3;" + Tc_BLUESPACE + "=3"
+	var/on = 0
+	var/cover_open = 0
+	var/base_state = "radio_jammer"
+	var/obj/item/weapon/cell/power_src = null
+	var/power_usage = 500 // approx 1 minute of leisure action on default cells
```

Comment:
actually process() is called about every 2 or 2.5 seconds, it will take 20 seconds for it to consume 5000 power, and in 30 seconds will consume a default power cell which is 7500. You might probably mean the high-capacity power cell

Reply:
yes, I meant the high-cap one like the ones in the tool storage because AFAIK there are no basic cells lying around on maps

Reply:
fair